nextjs-nabil/pages/_app.js

```diff
@@ -1,19 +1,19 @@
 import Layout from "../src/layouts/Layout";
 import "../styles/globals.scss";
-import LoginProvider from "../src/context";
+import AppProvider from "../src/context";
 import Head from 'next/head';
 
 function MyApp({ Component, pageProps }) {
   return (
-    <LoginProvider>
+    <AppProvider>
```

Comment:
It's okay to have a general context but it's better to have it split and not to handle every situation and it becomes easy to have it bloated and hard to manage. If it's only for authentication, then advisable to name it as such to provide clarity

Reply:
Noted